community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py

```diff
@@ -1784,19 +1818,35 @@ def nodeset_reservation(self, nodeset: object) -> Optional[ReservationDetails]:
             )
 
         project, name = match.group("project", "reservation")
-        reservation = self._get_reservation(project, zone, name)
+        return self.get_reservation_details(project, zone, name, nodeset.reservation_name)
+    
+    def future_reservation(self, nodeset:object) -> Optional[FutureReservation]:
+        if not nodeset.future_reservation:
+            return None
 
-        # Converts policy URLs to names, e.g.:
-        # projects/111111/regions/us-central1/resourcePolicies/zebra -> zebra
-        policies = [u.split("/")[-1] for u in reservation.get("resourcePolicies", {}).values()]
+        active_reservation = None
+        match = re.search(r'^projects/(?P<project>[^/]+)/zones/(?P<zone>[^/]+)/futureReservations/(?P<name>[^/]+)(/.*)?$', nodeset.future_reservation)
+        project, zone, name = match.group("project","zone","name")
+        fr = self._get_future_reservation(project,zone,name)
 
-        return ReservationDetails(
+        start_time = datetime.fromisoformat(fr["timeWindow"]["startTime"][:-1])
```

Comment:
Please add a TODO to remove trimming hack after moving to 3.11
Also you can provide context: https://discuss.python.org/t/parse-z-timezone-suffix-in-datetime/2220/30